From 3fbaecd433a1e1bbc96526c4f92d3d091325e4d8 Mon Sep 17 00:00:00 2001
From: Florent Auger <florent.auger@intel.com>
Date: Thu, 12 Mar 2015 18:52:03 +0100
Subject: [PATCH] Make build-able the intel_warn_int driver as module

A symbol had to be exported in hw_nmi.c to build the driver
as a module.
The driver was added to a Makefile to compile it once
enabled in the kernel config.
Also, corrected the comment in Kconfig about the timeout value
when the TCO timer is reloaded after the first expiration.

Change-Id: I609b69f18309da76a96af0c02a71f26df4e7b585
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1229
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-7015
Signed-off-by: Florent Auger <florent.auger@intel.com>
---
 arch/x86/kernel/apic/hw_nmi.c | 1 +
 drivers/platform/x86/Kconfig  | 4 ++--
 drivers/platform/x86/Makefile | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/apic/hw_nmi.c b/arch/x86/kernel/apic/hw_nmi.c
index a698d71..177c913 100644
--- a/arch/x86/kernel/apic/hw_nmi.c
+++ b/arch/x86/kernel/apic/hw_nmi.c
@@ -59,6 +59,7 @@ void arch_trigger_all_cpu_backtrace(void)
 	clear_bit(0, &backtrace_flag);
 	smp_mb__after_clear_bit();
 }
+EXPORT_SYMBOL(arch_trigger_all_cpu_backtrace);
 
 static int __kprobes
 arch_trigger_all_cpu_backtrace_handler(unsigned int cmd, struct pt_regs *regs)
diff --git a/drivers/platform/x86/Kconfig b/drivers/platform/x86/Kconfig
index ce8258b..1f607d4 100644
--- a/drivers/platform/x86/Kconfig
+++ b/drivers/platform/x86/Kconfig
@@ -835,10 +835,10 @@ config INTEL_WARN_INT
 	  Upon first expiration of the TCO watchdog, a warning interrupt is
 	  fired and the driver calls panic() function to dump debug information
 	  (registers and call stacks).
-	  At the same time, the TCO watchdog reloads with 2.4 seconds timeout
+	  At the same time, the TCO watchdog reloads with 4 seconds timeout
 	  value and runs till the second expiration. At the second expiration of
 	  the TCO watchdog, the platform resets (the dump is supposed to last less
-	  than 2.4 seconds).
+	  than 4 seconds).
 
 config INTEL_GPIO_HSIC
 	tristate "Intel HSIC GPIO support"
diff --git a/drivers/platform/x86/Makefile b/drivers/platform/x86/Makefile
index 38d342f..e9383dd 100644
--- a/drivers/platform/x86/Makefile
+++ b/drivers/platform/x86/Makefile
@@ -53,6 +53,7 @@ obj-$(CONFIG_SAMSUNG_Q10)	+= samsung-q10.o
 obj-$(CONFIG_APPLE_GMUX)	+= apple-gmux.o
 obj-$(CONFIG_INTEL_RST)		+= intel-rst.o
 obj-$(CONFIG_INTEL_SMARTCONNECT)	+= intel-smartconnect.o
+obj-$(CONFIG_INTEL_WARN_INT)	+= intel_warn_int.o
 
 obj-$(CONFIG_PVPANIC)		+= pvpanic.o
 obj-$(CONFIG_INTEL_BAYTRAIL_MBI)	+= intel_baytrail.o
-- 
1.9.1

